include/seqan3/core/type_traits/lazy.hpp

```diff
@@ -95,13 +95,50 @@ struct lazy_conditional : instantiate<std::conditional_t<decision, on_true_t, on
  * \tparam decision   Whether to resolve to the first type or the second.
  * \tparam on_true_t  The return type in case `decision` is true.
  * \tparam on_false_t The return type in case `decision` is false.
- * \relates seqan3::lazy_conditional
+ * \relates seqan3::detail::lazy_conditional
  */
 template <bool decision, typename on_true_t, typename on_false_t>
 //!\cond
     requires requires { typename instantiate_t<std::conditional_t<decision, on_true_t, on_false_t>>; }
 //!\endcond
 using lazy_conditional_t = instantiate_t<std::conditional_t<decision, on_true_t, on_false_t>>;
 
+/*!\brief An unary type trait that tests whether a template class can be instantiated with the given template type
+ *        parameters.
+ * \implements seqan3::unary_type_trait
+ * \tparam query_t The type of the template class to test.
+ * \tparam args_t  The template parameter pack to instantiate the template class with.
+ *
+ * \details
+ *
+ * Note, this unary type trait can be used in a seqan3::detail::lazy_conditional expression to check if instantiating
+ * a template class with specific template arguments would result in a valid template definition. Thus, the template
+ * parameters of the checked class must be constrained accordingly.
+ *
+ * ### Example
+ *
+ * \include test/snippet/core/type_traits/is_instantiable_with.cpp
+ */
+template <template <typename ...> typename query_t, typename ...args_t>
+struct is_instantiable_with :
+//!\cond
+    public std::false_type
+//!\endcond
+{};
+
+//!\cond
+template <template <typename ...> typename query_t, typename ...args_t>
+    requires requires { typename std::type_identity<query_t<args_t...>>::type; }
```

Comment:
Huh, I am a little confused. Whether a type can be instantiated with the given template arguments or whether the resulting type is incomplete is a different pair of shoes. This unary type trait is not supposed to tell me if I can form an instance of this class template. This becomes far more complex (if possible at all) since I don‘t know how I can form an instance of the class without introspection. I have to check if a SFINAE expression using declval can help. In addition, this is a detail helper function for our code base in which we wouldn‘t allow the baz use case anyhow. There are probably reasons when you want to have the type unconstrained like in some meta code, but I don‘t see that this would be a use case for this unary type trait. Let‘s discuss this in the office

Reply:
Ok as discussed I rename it to `is_class_template_declarable_with`